kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala

```diff
@@ -267,7 +267,44 @@ private[v1] class AdminResource extends ApiRequestContext with Logging {
       @QueryParam("type") engineType: String,
       @QueryParam("sharelevel") shareLevel: String,
       @QueryParam("subdomain") subdomain: String,
-      @QueryParam("hive.server2.proxy.user") hs2ProxyUser: String): Seq[Engine] = {
+      @QueryParam("hive.server2.proxy.user") hs2ProxyUser: String,
+      @QueryParam("allengine") allengine: Boolean): Seq[Engine] = {
+    if (allengine) {
+      val engines = ListBuffer[Engine]()
```

Comment:
When listing all engines, we need to check whether the user is an administrator.

https://github.com/apache/kyuubi/blob/beea6d5fce94043c29ed2681fc65e704f523a916/kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala#L331-L334